src/js/controllers/column-persistance-service.js

```diff
@@ -0,0 +1,150 @@
+import axios from 'axios'
+import debounce from 'lodash/debounce'
+import PersistanceService from './persistance-service.js'
+
+/* istanbul ignore next */
+export default class ColumnPersistanceService extends PersistanceService {
+  constructor(column, scroll_version_id, session_id) {
+    super()
+
+    this.scroll_version_id = scroll_version_id
+    this.session_id = session_id
+
+    this.column = column
+
+    this.column.on(['change', 'addition', 'delete'], this.onChange.bind(this))
+  }
+
+  /**
+   * Actually make the request to persist any tracked changes
+   *
+   * @protected
+   * @instance
+   *
+   * @param {object} onDeckChanges The changes to persist
+   */
+  _makeRequest(onDeckChanges) {
+    const transactions = this._gatherTransactions(onDeckChanges)
+
+    // Make the request if there are transactions to send.
+    if (transactions.length) {
+      const requests = transactions.reduce((requests, transaction, i) => {
+        requests[`${i + 1}`] = transaction
+        return requests
+      }, {})
+      return axios
+        .post('resources/cgi-bin/scrollery-cgi.pl', {
+          SESSION_ID: this.session_id,
+          requests,
+        })
+        .then(res => {
+          if (res.status === 200 && res.data) {
+            this.onSuccess(requests, res.data.replies)
+          }
+        })
+        .catch(res => {
+          res
+        })
+    } else {
+      // there are no things to save ... return a promise that
+      // resolves immediately to adhere to the interface
+      return new Promise(r => r())
+    }
+  }
+
+  /**
+   * @private
+   * @instance
+   */
+  _onSuccess(transactions, replies) {
+    const persistedMap = {
+      additions: {},
+      deletions: {},
+      updates: {},
+    }
+    for (let id in replies) {
+      const transaction = transactions[id]
+      replies[id].forEach((singleAction, i) => {
+        for (let signUuid in singleAction) {
+          // the value is the keyword 'deleted', then the corresponding
+          // sign has been removed.
+          if (singleAction[signUuid] === 'deleted') {
+            persistedMap.deletions[signUuid] = true
+
+            // if the value is a number, it's the sign's id after being persisted.
+          } else if (typeof singleAction[signUuid] === 'number') {
```

Comment:
I don't know the answer to this, but I assume you are right that 'number' with a lower case 'n' will work correctly even though I am generally familiar with it as a 'Number'.  For my benefit, is this the best way to test for object type?  I have had trouble finding a definitive answer on that.  Thanks.